docs/design-patterns.md

```diff
@@ -474,3 +474,38 @@ transaction {
     }
 }
 ```
+
+## Capability Revocation
+
+### Problem
+
+A capability provided by one account to a second account must able to be revoked
+by the first account without the co-operation of the second.
+
+### Solution
+
+If the capability is a storage capability:
+
+```cadence
+transaction(capabilityID: UInt64) {
+    prepare(signer: auth(StorageCapabilities) &Account) {
+        let controller = signer.capabilities.storage
+            .getController(byCapabilityID: capabilityID)
+            ?? panic("missing controller")
```

Comment:
```suggestion
            ?? panic("Cannot get the storage capability controller with ID "
                    .concat(capabilityID.toString())
                    .concat(" from the signer's account! Make sure the ID belongs to a capability that the owner controls and that it is a storage capability.")
```

We're trying to go through and make sure that all the error messages in all our code are more helpful so it would be good to do that for all new code that we add from now on, even examples.